### ISSUE TITLE
How to create PDF and edit button on DataTables?

### ISSUE THREAD

**Issue by wachara-yanu** [diedona/DDona.Datatables#1]
Could you please share your code how to create PDF export and edit button on datatables jquery?

**Comment by diedona** [comment]
@watcharayarnu hello, i can put on an stub example later, but in this project i didn't really made any pdf or edit features.

Later i will open the project and put on the stubs for what you asked, is that ok?

**Comment by wachara-yanu** [comment]
ok. I'm very pleasure. nice to know you @diedona

**Comment by diedona** [comment]
@watcharayarnu nice to meet you too. I've already commited the changes, see if it can help you.

**Comment by watcharayarnu** [comment]
Very cool !  can it export to pdf file on click at pdf button and uri link to edit function on click at edit button.

**Comment by wachara-yanu** [comment]
@diedona Very cool !  can it export to pdf file on click at pdf button and uri link to edit function on click at edit button.

**Comment by diedona** [comment]
@watcharayarnu not exactly... Since it's a server side datatables, you never have all the data in the HTML. So you need to program the PDF button to query ALL data and then export it to pdf! Not so straight away, but it's possible.

As for the edit, the most common is to redirect to an edit page with the Id of the current row (you can get it from the data row in the javascript).

It's just a matter of filling in the blanks :smile:

**Comment by wachara-yanu** [comment]
@diedona  Could you have a new commit the changes?

**Comment by diedona** [comment]
@watcharayarnu sorry, the code for these implementations go beyond my stub... it would take a certain time to write a full edit/pdf features and i am kinda busy in a new personal project. Can you go on from this point?

**Comment by wachara-yanu** [comment]
ok! @diedona I can go on from this point and hope to see a full edit/pdf features from you.

**Comment by wachara-yanu** [comment]
@diedona do you know how to draw a chart js from json data? please suggest me.

**Comment by diedona** [comment]
@vajrayanu hmm... i would suggest HighCharts (http://www.highcharts.com/) it's tricky, but it's the least tricky i know

**Comment by wachara-yanu** [comment]
@diedona Thank you so much. It's very good chart but now I used http://www.chartjs.org/

I've one question to ask you. suppose I have json data  10 records and I need to separate it to 10 charts per one page.
How to do it? can you suggest me?

**Comment by diedona** [comment]
@vajrayanu it seems tricky to answer, for there are many ways to accomplish it... You could use each index (json[0], json[1], json[x]) to be the datasource of each chart, but i'm not sure about your data structure...

**Comment by wachara-yanu** [comment]
@diedona ok I think so.
